test/unit/transports/local_test.rb

```diff
@@ -6,10 +6,11 @@
 class TransportHelper
   attr_accessor :transport
 
-  def initialize
+  def initialize(type = :mock)
     Train::Platforms::Detect::Specifications::OS.load
-    plat = Train::Platforms.name('mock').in_family('linux')
+    plat = Train::Platforms.name('mock')
     plat.add_platform_methods
+    plat.stubs(:windows?).returns(true) if type == :windows
```

Comment:
The seems a bit brute-force. Can we accept an options hash and then we can plumb the family in as necessary and let the `Train::Platforms` stuff work normally without stubbing methods on this?

```ruby
def initialize(opts = {})
  Train::Platforms::Detect::Specifications::OS.load
  plat = Train::Platforms.name('mock')
  plat.in_family(opts[:family]) unless opts[:family].nil?
  plat.add_platform_methods
end
```

Would that work?

Reply:
Sure will! Great idea. See latest commit.